velox/functions/lib/QuantileDigest.h

```diff
@@ -1371,6 +1386,77 @@ QuantileDigest<T, Allocator>::getDistributionFunction(
   return cdf;
 }
 
+template <typename T, typename Allocator>
+template <typename ValueFunc>
+std::optional<double> QuantileDigest<T, Allocator>::getTruncatedMean(
```

Comment:
To make this API more general, how about making it return a list of values and their nodeCounts for values within the [lowerQuantileBound, upperQuantileBound] range, and let the UDF decide what it would compute (e.g., mean, medium, stddev, etc.)?

Reply:
That's a very good idea. I did some refactoring so the other statistical types can use the core logic here. I've still kept the UDF straightforward and development isn't planned on TDigest/QDigest data types, it's mostly for existing workloads. How do the new changes look?